packages/core/src/response.ts

```diff
@@ -90,7 +90,9 @@ export class Response {
     }
 
     if (fireInvalidEvent(response)) {
-      return modal.show(response.data)
+      if (modal.getConfig().showOnNonInertiaResponse) {
```

Comment:
maybe this logic could be moved inside the `modal.show` so the caller doesn't need to know about this config or any business rule